src/middleware/rate-limiter.ts

```diff
@@ -0,0 +1,13 @@
+import rateLimit from 'express-rate-limit';
+
+export const rateLimiter = rateLimit({
+    windowMs: parseInt(process.env.RATE_LIMIT_WINDOW_MS || '60000', 10),
+    max: parseInt(process.env.RATE_LIMIT_MAX_REQ || '100', 10),
+    standardHeaders: true,
+    legacyHeaders: false,
+    handler: (req, res) => {
+        res.status(429).json({
+            message: 'Too many requests, please try again later.'
+        });
+    }
+});
```

Comment:
GitHub was complaining about no rate limiter on the API, so I copied the middleware from the frontend.

Reply:
Seems sensible to me... We do need to keep the code sniffer happy generally.